### ISSUE TITLE
Detect Indentation failing on 3 and 5 space indented C files

### ISSUE THREAD

**Issue by AiryShift** [Microsoft/vscode#31921]
- VSCode Version: Code 1.14.2 (cb82febafda0c8c199b9201ad274e25d9a76874e, 2017-07-19T23:26:08.116Z)
- OS Version: Linux x64 4.11.11-300.fc26.x86_64
- Extensions: 

Extension|Author (truncated)|Version
---|---|---
vscode-markdownlint|Dav|0.10.0
latex-workshop|Jam|3.2.0
material-icon-theme|PKi|2.0.11
code-settings-sync|Sha|2.8.2
python|don|0.6.9
MagicPython|mag|1.0.11
cpptools|ms-|0.12.1
sublime-keybindings|ms-|2.9.1
expand-selection-to-scope|vit|0.2.0

(1 theme extensions excluded)

Setting `editor.detectIndentation` as well as Command Palette command "Detect Indentation from Content" fails to work on 3 and 5 space indented C files. 

Example:
```c
int main(void) {
   int i;
   for (i = 0; i < 10; i++) {
      printf("%d\n", i);
   }
}
```
Here indent size is incorrectly set to 4 when it should be 3.

Example:
```c
int f(int x)
{
   if (10 <= x && x <= 20)
      return x;
   else
      return -x;
}
```
Here indent size is incorrectly set to 6 when it should be 3.

Example:
```c
#include <stdio.h>

int f(int x)
{
     if (10 <= x && x <= 20)
          return x;
     else
          return -x;
}

int main(void) {
     int i;
     for (i = 0; i < 10; i++) {
          printf("%d\n", f(i));
     }
     return 0;
}
```
Here indent size is incorrectly set to 4 when it should be 5.

Steps to Reproduce:

1. Open new file
2. Set Language Mode to C
3. Paste one of the code snippets
4. Run Command Palette command "Detect Indentation from Content"

Reproduces without extensions: Yes

***
Also reproduced on a fresh install of the insider build:
- VSCode Version: Code - Insiders 1.15.0-insider (211ffdf184329179ddca9c1fe993f5f11f3c4dda, 2017-08-02T05:09:50.364Z)
- OS Version: Linux x64 4.11.11-300.fc26.x86_64
- Extensions: none

**Comment by AiryShift** [comment]
I found in the file `/vscode/src/vs/editor/common/model/indentationGuesser.ts` (file that contains the implementation for the `detectIndentation` functionality) the line
```ts
const ALLOWED_TAB_SIZE_GUESSES = [2, 4, 6, 8];	// limit guesses for `tabSize` to 2, 4, 6 or 8.
```
Is there any reason why only even numbered guesses are allowed? This seems counter-intuitive as 3 spaced tab sizes are not unheard of, as well as Code itself offering custom tab sizes from 1 to 8.

**Comment by AiryShift** [comment]
Closed as is duplicate of #17639
